.storybook/utils.ts

```diff
@@ -40,6 +40,23 @@ export const disableStorybookControls = <Props>(
 };
 const DISABLE_CONTROL = { control: false };
 
+/**
+ * Configure provided args to be listed under a specified
+ * category in the props table.
+ *
+ * Should be passed or spread to `argTypes`
+ */
+export const moveStorybookControlsToCategory = <Props>(
+  propNames: Array<keyof Props>,
+  category = 'Additional'
+): Record<keyof Props, ControlCategory> | {} => {
+  return propNames.reduce(
+    (obj, name) => ({ ...obj, [name]: { table: { category } } }),
```

Comment:
Oooh actually, something I just realized - we're going to need to modify all our utils to merge existing props instead of overriding the entire obj. For example, we can't **both** hide/disable a control **and** move it right now because one overrides the other completely. You can see this in [EuiFacetButton's story](https://eui.elastic.co/pr_7569/storybook/index.html?path=/story/euifacetbutton--playground) where `buttonRef` should be both disabled and moved but isn't(?).

```suggestion
    (obj, name) => ({ ...obj, [name]: { ...obj[name], table: { ...obj[name]?.table category } } }),
```

LMK if that makes sense - I'm possibly not explaining it super clearly, but happy to chat in Slack etc if I'm not 😅

Reply:
I think currently the `buttonRef` in `EuiFacetButton` is not moved or disabled yet. But I get the issue anyway! 👀 

What works currently and why it really show up as a problem for me yet: 
* when using a ...spread util in the component `argTypes` and a ...spread util in the story `argTypes` it seems to be merged by storybook
* when the util in the component `argTypes` is not ...spread it won't be merged
* when using two utils in the component `argTypes` (even when ...spread) the last one overwrites anything before

Reply:
Actually I think we need to think about this a bit more, because how it currently works actually allow for merging in the utils because we only work on basis of the passed props for which we generate an object. But for each util this is scoped and it has no context of other utils outputs.

```js
// component argTypes
moveStorybookControlsToCategory(['buttonRef'])

// output
{
    "buttonRef": {
        "table": {
            "category": "EuiButton props"
        }
    }
}

// story argTypes (overwrite)
disableStorybookControls(['buttonRef'])

// output - overwrites the component argTypes as the object key is the same
{
    "buttonRef": {
        "control": false
    }
}
```

For now it works if we define usage as follows:
- we **can't** use multiple utils for the same prop in the the component `argTypes`
- we **can't** use multiple utils for the same prop in the the story `argTypes`
- we can define disable/hide OR move util for the same prop in the component `argTypes` and the other in the story `argTypes` (or vice versa)

Reply:
Yeah, I would also want to abstract that imperative mess away 😄 I'm not the most happy about mutating the config objects if we don't return them but we're kinda locked in by how Storybook handles this with the CSF 🤷‍♀️ 

We'd still need to make sure that multiple calls of different utils work as expected but I think this could be a good solution to explore. That being said - I'd suggest to move this update out of this PR and into its own issue to handle separately. Does that sound good to you?

Reply:
Totally good with that!

Reply:
📔 I created a separate [issue](https://github.com/elastic/eui/issues/7583) for this.